Remove ocp-prod apps from this repository

larsks opened the pull request:
We would like to manage the app-of-apps configuration for the ocp-prod cluster through a different repository. See [1] for our proposed configuration.

[1]: https://github.com/CCI-MOC/moc-apps/pull/38

I'd like to make sure we do this in a way that doesn't result in all of resources getting deleted. Assuming that this PR by itself isn't sufficient and hoping for feedback on the correct solution :smile: .

HumairAK commented:
This PR would not delete any resources. In fact for some reason right now argocd does not correctly remove applications during sync (see [here](https://github.com/operate-first/SRE/issues/378)), but even if it deleted the apps, it doesn't "cascade" delete them i.e. delete everything deployed by those apps, though the option to do that (at least with the ui) is there.

HumairAK commented on app-of-apps/app-of-apps-ocp-prod.yaml:
```diff
@@ -1,20 +0,0 @@
-apiVersion: argoproj.io/v1alpha1
```
Any particular reason you want the apps to live in a separate org? 

If you remove this app, you need to remove this line as well: https://github.com/operate-first/argocd-apps/blob/main/app-of-apps/kustomization.yaml#L8 

But if you do this, then all these manifests are unreachable by argocd. Since ArgoCD is deploying all the apps within the `app-of-apps` directory, which in turn deploy the rest of the apps. 

What you _could_ do is point this line: https://github.com/operate-first/argocd-apps/blob/main/app-of-apps/kustomization.yaml#L8 to the remote location of your app-of-apps. 

The security concern here is that this org is not part of operate-first, so you could technically create an app that belongs to a different project, namespace, cluster, etc. w/o our oversight (during PR review phase). 

This was the main reason why we forced everyone to put their apps declaratively in this repo [see here](https://github.com/operate-first/blueprint/blob/main/docs/adr/0004-argocd-apps-of-apps-structure.md#negative-consequences). But since then [this pr has been merged](https://github.com/argoproj/argo-cd/issues/3045), so maybe it's time to revisit this.

HumairAK replied:
So my understanding is, we would need to do the following to prevent the above: 

in this [file](https://github.com/operate-first/argocd-apps/blob/main/projects/global_project.yaml) add: 

```
  sourceRepos:
    - 'https://github.com/operate-first/*'
```
This is our global project, so by default all projects should now only be allowed to deploy from operate-first org. 

Then update individual projects for the repos they are permitted to source from.

This can be a separate PR. I can do this. WDYT?

larsks replied:
> but even if it deleted the apps, it doesn't "cascade" delete them 

Well, that's a relief :)

> But if you do this, then all these manifests are unreachable by argocd.

Right, I was assuming we would deploy our own top-level app-of-apps application. But...

> What you could do is point this line: https://github.com/operate-first/argocd-apps/blob/main/app-of-apps/kustomization.yaml#L8 to the remote location of your app-of-apps.

...we could try that.

> This can be a separate PR. I can do this. WDYT?

That sounds like a good idea in general.

larsks replied:
It looks like we're setting:

```
sourceRepos:
  - '*'
```

In many of the projects (including `mocops`). I guess if we make the change to `global-project.yaml`, the `mocops` project will work without modification  (it would be able to fetch from `https://github/com/CCI-MOC/...` because that matches `*`).

HumairAK replied:
@larsks what we want to do I think is remove `*` in all these projects and be more explicit in what repos each project is allowed to source from. So I'm thinking for mocops change https://github.com/larsks/argocd-apps/blob/feature/move-ocp-prod-apps/projects/mocops.yaml#L10 to `https://github/com/CCI-MOC/*`  that way any app belonging to the `mocop` project can source from operate-first org (inherited from global project) and the cci-moc org, wdyt? 

then I'll follow up another PR changing the rest of the projects, as some of them source from other orgs like `aicoe` `thoth` etc. which is kinda unrelated to this pr.

larsks replied:
Sure, will do.

larsks committed:
Limit projects by default to operate first repositories

Per @HumairAK's comments in [1].

[1]: https://github.com/operate-first/argocd-apps/pull/196

larsks committed:
Move MOC ocp-prod app-of-apps to MOC repository

We would like to manage the app-of-apps configuration for the ocp-prod
cluster through a different repository. See 1 for our proposed
configuration.

[1]: https://github.com/CCI-MOC/moc-apps/pull/38

sesheta commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **4n4nd** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @4n4nd` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fargocd-apps).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/operate-first/argocd-apps/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["4n4nd"]} -->

sesheta commented:
@larsks: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
aicoe-ci/prow/pre-commit | d0f189902d4a98381452f627ac405e073a92d118 | [link](https://prow.operate-first.cloud/view/s3/ci-prow-artifacts--a96a53d7-728c-4ebb-bfc4-2f9e4dc68070/prow-logs/pr-logs/pull/operate-first_argocd-apps/196/pre-commit/1435978722917224448) | `/test pre-commit`
aicoe-ci/prow/kustomize-build | d0f189902d4a98381452f627ac405e073a92d118 | [link](https://prow.operate-first.cloud/view/s3/ci-prow-artifacts--a96a53d7-728c-4ebb-bfc4-2f9e4dc68070/prow-logs/pr-logs/pull/operate-first_argocd-apps/196/kustomize-build/1435978719356260352) | `/test kustomize-build`

[Full PR test history](https://prow.operate-first.cloud/pr-history?org=operate-first&repo=argocd-apps&pr=196). [Your PR dashboard](https://prow.operate-first.cloud/pr?query=is%3Apr%20state%3Aopen%20author%3Alarsks). Please help us and [open an issue](https://github.com/operate-first/argocd-apps/issues?q=is:issue+is:open) when you hit one in your PR.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

sesheta commented:
@larsks: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
aicoe-ci/prow/pre-commit | d0f189902d4a98381452f627ac405e073a92d118 | [link](https://prow.operate-first.cloud/view/s3/ci-prow-artifacts--a96a53d7-728c-4ebb-bfc4-2f9e4dc68070/prow-logs/pr-logs/pull/operate-first_argocd-apps/196/pre-commit/1435978722917224448) | `/test pre-commit`
aicoe-ci/prow/kustomize-build | dcc137d97a1a608d84fb0ceb7470fa30975550fe | [link](https://prow.operate-first.cloud/view/s3/ci-prow-artifacts--a96a53d7-728c-4ebb-bfc4-2f9e4dc68070/prow-logs/pr-logs/pull/operate-first_argocd-apps/196/kustomize-build/1435979229161328640) | `/test kustomize-build`

[Full PR test history](https://prow.operate-first.cloud/pr-history?org=operate-first&repo=argocd-apps&pr=196). [Your PR dashboard](https://prow.operate-first.cloud/pr?query=is%3Apr%20state%3Aopen%20author%3Alarsks). Please help us and [open an issue](https://github.com/operate-first/argocd-apps/issues?q=is:issue+is:open) when you hit one in your PR.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

sesheta commented:
@larsks: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
aicoe-ci/prow/kustomize-build | dcc137d97a1a608d84fb0ceb7470fa30975550fe | [link](https://prow.operate-first.cloud/view/s3/ci-prow-artifacts--a96a53d7-728c-4ebb-bfc4-2f9e4dc68070/prow-logs/pr-logs/pull/operate-first_argocd-apps/196/kustomize-build/1435979229161328640) | `/test kustomize-build`
aicoe-ci/prow/pre-commit | dcc137d97a1a608d84fb0ceb7470fa30975550fe | [link](https://prow.operate-first.cloud/view/s3/ci-prow-artifacts--a96a53d7-728c-4ebb-bfc4-2f9e4dc68070/prow-logs/pr-logs/pull/operate-first_argocd-apps/196/pre-commit/1435979229383626752) | `/test pre-commit`

[Full PR test history](https://prow.operate-first.cloud/pr-history?org=operate-first&repo=argocd-apps&pr=196). [Your PR dashboard](https://prow.operate-first.cloud/pr?query=is%3Apr%20state%3Aopen%20author%3Alarsks). Please help us and [open an issue](https://github.com/operate-first/argocd-apps/issues?q=is:issue+is:open) when you hit one in your PR.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

larsks commented:
/unhold

harshad16 commented:
/test all

larsks commented:
/hold

Putting this on hold briefly until some add'l changes merge into the moc-apps repo.

larsks commented:
/unhold

HumairAK commented:
/lgtm
/approve

HumairAK commented:
/retest
